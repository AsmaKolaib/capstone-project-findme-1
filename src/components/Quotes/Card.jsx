import React from 'react'
//importing icons from image folder
import qoute from '../../../src/images/Quote.svg'
//we used CSS file to cusomize its styles 
import './test.css'

export default function Card({cardInfo,i}) {
   // console.log({this.props.cardInfo})
   //Tailwind classes are used to style the code elements
   //test.css is used to customize slider buttons classes properties 
   // Each card in the slider will have image, header, and paragragh
   //data is recieved as props and displayed in the card 
    return (
        <div style={ cardContainer2} className='bg-findMe p-0 m-0 justify-items-center flex flex-col items-center  sm:flex-row-reverse '>
            <div className='p-4 xl:h-full1 xl:w-3/4 lg:w-3/4 lg:h-full11 md:h-full2 md:w-3/4 sm:w-3/4 sm:h-full3 h-full4 w-full flex-1   rounded-l-3xl rounded-r-3xl  ' >
                <img src={cardInfo.img}  alt='card ' className=' object-cover h-full w-full rounded-l-3xl rounded-r-3xl'/>
            </div>
            <div className='items-center flex flex-col items-center  flex-1 sm:items-start w:3/4  pt-0'>
                <div className='flex space-x-1 flex-1  self-end flex sm:mr-8 sm:mt-8   '>
                    <img src={qoute} alt='flex-1 quotes icons' className='flex-1 pl-10 pt-1 md:w-16 md:h-16 xl:w-20 xl:h-20 sm:pl-10 lg:w-20 lg:h-20 w-16 h-16 '></img>
                    <img src={qoute} alt='flex-1 quotes icons' className='flex-1 pr-10 pt-1  md:w-16 md:h-16  lg:w-20 lg:h-20 xl:w-20 xl:h-20 w-16 h-16 sm:pr-10 '></img>

                </div>
                <h1 className='pb-4 text-blue-500 sm:text-base md:text-base self-center sm:self-start xl:text-2xl lg:text-2xl   flex-1 items-start bold font-bold '>{cardInfo.head}</h1>
                <p className='flex-1 sm:left text-center sm:text-left leading-normal sm:leading-snug md:leading-snug lg:leading-loose xl:leading-loose lg:text-base xl:text-base md:text-sm sm:text-xs text-xs  w-4/5 xl:w-3/4	'>{cardInfo.para}</p>
            </div>
        </div>
    )
}

const cardContainer2 = {
    paddingLeft: '5%',
    paddingRight: '5%',
    
}