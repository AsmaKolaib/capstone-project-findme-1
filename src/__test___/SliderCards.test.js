import { cleanup } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
//import { isTSAnyKeyword } from '@babel/types';
import { render } from '@testing-library/react'
import Slider from '../../../containers/Home/SliderCards';
import Card from '../Card'

import renderer from 'react-test-renderer';

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render( < Slider / > , div);
});

it('matches snapshot', () => {
    const tree = renderer.create( < Slider / > ).toJSON();
    expect(tree).toMatchSnapshot();
});