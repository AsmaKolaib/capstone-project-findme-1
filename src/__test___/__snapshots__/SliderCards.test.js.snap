// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className=" bg-findMe max-h-full sm:pr-4 sm:pl-4 "
>
  <div
    className="sc-jSFkmK llldhF rec rec-carousel-wrapper "
    style={Object {}}
  >
    <div
      className="sc-eCApGN cfKeGQ rec rec-carousel"
      size={
        Object {
          "height": 0,
        }
      }
      style={
        Object {
          "height": 0,
        }
      }
    >
      <button
        className="sc-bdnylx hCVuYx rec rec-arrow rec rec-arrow-left"
        disabled={true}
        onClick={[Function]}
        style={Object {}}
        tabIndex={0}
        type="button"
      >
        ‚ùÆ
      </button>
      <div
        className="sc-dlnjPT mUZZN rec rec-slider-container"
      >
        <div
          className="sc-hKFyIo cIpIiu rec rec-slider"
        >
          <div
            className="rec rec-swipable"
            onMouseDown={[Function]}
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <div
              className="rec rec-carousel-item rec-carousel-item-0"
            >
              <div
                className="sc-gtssRu iamfxf rec rec-item-wrapper"
                id={0}
                onClick={[Function]}
                style={
                  Object {
                    "padding": "0px 0px 0px 0px",
                    "width": "0px",
                  }
                }
              >
                <div
                  className="bg-findMe p-0 m-0 justify-items-center flex flex-col items-center  sm:flex-row-reverse "
                  style={
                    Object {
                      "paddingLeft": "5%",
                      "paddingRight": "5%",
                    }
                  }
                >
                  <div
                    className="p-4 xl:h-full1 xl:w-3/4 lg:w-3/4 lg:h-full11 md:h-full2 md:w-3/4 sm:w-3/4 sm:h-full3 h-full4 w-full flex-1   rounded-l-3xl rounded-r-3xl  "
                  >
                    <img
                      alt="card "
                      className=" object-cover h-full w-full rounded-l-3xl rounded-r-3xl"
                      src="pic1.jpg"
                    />
                  </div>
                  <div
                    className="items-center flex flex-col items-center  flex-1 sm:items-start w:3/4  pt-0"
                  >
                    <div
                      className="flex space-x-1 flex-1  self-end flex sm:mr-8 sm:mt-8   "
                    >
                      <img
                        alt="flex-1 quotes icons"
                        className="flex-1 pl-10 pt-1 md:w-16 md:h-16 xl:w-20 xl:h-20 sm:pl-10 lg:w-20 lg:h-20 w-16 h-16 "
                        src="Quote.svg"
                      />
                      <img
                        alt="flex-1 quotes icons"
                        className="flex-1 pr-10 pt-1  md:w-16 md:h-16  lg:w-20 lg:h-20 xl:w-20 xl:h-20 w-16 h-16 sm:pr-10 "
                        src="Quote.svg"
                      />
                    </div>
                    <h1
                      className="pb-4 text-blue-500 sm:text-base md:text-base self-center sm:self-start xl:text-2xl lg:text-2xl   flex-1 items-start bold font-bold "
                    >
                      Person Name 1
                    </h1>
                    <p
                      className="flex-1 sm:left text-center sm:text-left leading-normal sm:leading-snug md:leading-snug lg:leading-loose xl:leading-loose lg:text-base xl:text-base md:text-sm sm:text-xs text-xs  w-4/5 xl:w-3/4	"
                    >
                      This website has helped me find my son, I will always be thankful for that.This website has helped me find my son, I will always be thankful for that.
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="rec rec-carousel-item rec-carousel-item-1"
            >
              <div
                className="sc-gtssRu iamfxf rec rec-item-wrapper"
                id={1}
                onClick={[Function]}
                style={
                  Object {
                    "padding": "0px 0px 0px 0px",
                    "width": "0px",
                  }
                }
              >
                <div
                  className="bg-findMe p-0 m-0 justify-items-center flex flex-col items-center  sm:flex-row-reverse "
                  style={
                    Object {
                      "paddingLeft": "5%",
                      "paddingRight": "5%",
                    }
                  }
                >
                  <div
                    className="p-4 xl:h-full1 xl:w-3/4 lg:w-3/4 lg:h-full11 md:h-full2 md:w-3/4 sm:w-3/4 sm:h-full3 h-full4 w-full flex-1   rounded-l-3xl rounded-r-3xl  "
                  >
                    <img
                      alt="card "
                      className=" object-cover h-full w-full rounded-l-3xl rounded-r-3xl"
                      src="pic2.png"
                    />
                  </div>
                  <div
                    className="items-center flex flex-col items-center  flex-1 sm:items-start w:3/4  pt-0"
                  >
                    <div
                      className="flex space-x-1 flex-1  self-end flex sm:mr-8 sm:mt-8   "
                    >
                      <img
                        alt="flex-1 quotes icons"
                        className="flex-1 pl-10 pt-1 md:w-16 md:h-16 xl:w-20 xl:h-20 sm:pl-10 lg:w-20 lg:h-20 w-16 h-16 "
                        src="Quote.svg"
                      />
                      <img
                        alt="flex-1 quotes icons"
                        className="flex-1 pr-10 pt-1  md:w-16 md:h-16  lg:w-20 lg:h-20 xl:w-20 xl:h-20 w-16 h-16 sm:pr-10 "
                        src="Quote.svg"
                      />
                    </div>
                    <h1
                      className="pb-4 text-blue-500 sm:text-base md:text-base self-center sm:self-start xl:text-2xl lg:text-2xl   flex-1 items-start bold font-bold "
                    >
                      Person Name 2
                    </h1>
                    <p
                      className="flex-1 sm:left text-center sm:text-left leading-normal sm:leading-snug md:leading-snug lg:leading-loose xl:leading-loose lg:text-base xl:text-base md:text-sm sm:text-xs text-xs  w-4/5 xl:w-3/4	"
                    >
                      This website has helped me find my son, I will always be thankful for that.
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="rec rec-carousel-item rec-carousel-item-2"
            >
              <div
                className="sc-gtssRu iamfxf rec rec-item-wrapper"
                id={2}
                onClick={[Function]}
                style={
                  Object {
                    "padding": "0px 0px 0px 0px",
                    "width": "0px",
                  }
                }
              >
                <div
                  className="bg-findMe p-0 m-0 justify-items-center flex flex-col items-center  sm:flex-row-reverse "
                  style={
                    Object {
                      "paddingLeft": "5%",
                      "paddingRight": "5%",
                    }
                  }
                >
                  <div
                    className="p-4 xl:h-full1 xl:w-3/4 lg:w-3/4 lg:h-full11 md:h-full2 md:w-3/4 sm:w-3/4 sm:h-full3 h-full4 w-full flex-1   rounded-l-3xl rounded-r-3xl  "
                  >
                    <img
                      alt="card "
                      className=" object-cover h-full w-full rounded-l-3xl rounded-r-3xl"
                      src="pic1.jpg"
                    />
                  </div>
                  <div
                    className="items-center flex flex-col items-center  flex-1 sm:items-start w:3/4  pt-0"
                  >
                    <div
                      className="flex space-x-1 flex-1  self-end flex sm:mr-8 sm:mt-8   "
                    >
                      <img
                        alt="flex-1 quotes icons"
                        className="flex-1 pl-10 pt-1 md:w-16 md:h-16 xl:w-20 xl:h-20 sm:pl-10 lg:w-20 lg:h-20 w-16 h-16 "
                        src="Quote.svg"
                      />
                      <img
                        alt="flex-1 quotes icons"
                        className="flex-1 pr-10 pt-1  md:w-16 md:h-16  lg:w-20 lg:h-20 xl:w-20 xl:h-20 w-16 h-16 sm:pr-10 "
                        src="Quote.svg"
                      />
                    </div>
                    <h1
                      className="pb-4 text-blue-500 sm:text-base md:text-base self-center sm:self-start xl:text-2xl lg:text-2xl   flex-1 items-start bold font-bold "
                    >
                      Person Name 4
                    </h1>
                    <p
                      className="flex-1 sm:left text-center sm:text-left leading-normal sm:leading-snug md:leading-snug lg:leading-loose xl:leading-loose lg:text-base xl:text-base md:text-sm sm:text-xs text-xs  w-4/5 xl:w-3/4	"
                    >
                      This website has helped me find my son, I will always be thankful for that.
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="rec rec-carousel-item rec-carousel-item-3"
            >
              <div
                className="sc-gtssRu iamfxf rec rec-item-wrapper"
                id={3}
                onClick={[Function]}
                style={
                  Object {
                    "padding": "0px 0px 0px 0px",
                    "width": "0px",
                  }
                }
              >
                <div
                  className="bg-findMe p-0 m-0 justify-items-center flex flex-col items-center  sm:flex-row-reverse "
                  style={
                    Object {
                      "paddingLeft": "5%",
                      "paddingRight": "5%",
                    }
                  }
                >
                  <div
                    className="p-4 xl:h-full1 xl:w-3/4 lg:w-3/4 lg:h-full11 md:h-full2 md:w-3/4 sm:w-3/4 sm:h-full3 h-full4 w-full flex-1   rounded-l-3xl rounded-r-3xl  "
                  >
                    <img
                      alt="card "
                      className=" object-cover h-full w-full rounded-l-3xl rounded-r-3xl"
                      src="pic2.png"
                    />
                  </div>
                  <div
                    className="items-center flex flex-col items-center  flex-1 sm:items-start w:3/4  pt-0"
                  >
                    <div
                      className="flex space-x-1 flex-1  self-end flex sm:mr-8 sm:mt-8   "
                    >
                      <img
                        alt="flex-1 quotes icons"
                        className="flex-1 pl-10 pt-1 md:w-16 md:h-16 xl:w-20 xl:h-20 sm:pl-10 lg:w-20 lg:h-20 w-16 h-16 "
                        src="Quote.svg"
                      />
                      <img
                        alt="flex-1 quotes icons"
                        className="flex-1 pr-10 pt-1  md:w-16 md:h-16  lg:w-20 lg:h-20 xl:w-20 xl:h-20 w-16 h-16 sm:pr-10 "
                        src="Quote.svg"
                      />
                    </div>
                    <h1
                      className="pb-4 text-blue-500 sm:text-base md:text-base self-center sm:self-start xl:text-2xl lg:text-2xl   flex-1 items-start bold font-bold "
                    >
                      Person Name 3
                    </h1>
                    <p
                      className="flex-1 sm:left text-center sm:text-left leading-normal sm:leading-snug md:leading-snug lg:leading-loose xl:leading-loose lg:text-base xl:text-base md:text-sm sm:text-xs text-xs  w-4/5 xl:w-3/4	"
                    >
                      This website has helped me find my son, I will always be thankful for that.
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="rec rec-carousel-item rec-carousel-item-4"
            >
              <div
                className="sc-gtssRu iamfxf rec rec-item-wrapper"
                id={4}
                onClick={[Function]}
                style={
                  Object {
                    "padding": "0px 0px 0px 0px",
                    "width": "0px",
                  }
                }
              >
                <div
                  className="bg-findMe p-0 m-0 justify-items-center flex flex-col items-center  sm:flex-row-reverse "
                  style={
                    Object {
                      "paddingLeft": "5%",
                      "paddingRight": "5%",
                    }
                  }
                >
                  <div
                    className="p-4 xl:h-full1 xl:w-3/4 lg:w-3/4 lg:h-full11 md:h-full2 md:w-3/4 sm:w-3/4 sm:h-full3 h-full4 w-full flex-1   rounded-l-3xl rounded-r-3xl  "
                  >
                    <img
                      alt="card "
                      className=" object-cover h-full w-full rounded-l-3xl rounded-r-3xl"
                      src="pic1.jpg"
                    />
                  </div>
                  <div
                    className="items-center flex flex-col items-center  flex-1 sm:items-start w:3/4  pt-0"
                  >
                    <div
                      className="flex space-x-1 flex-1  self-end flex sm:mr-8 sm:mt-8   "
                    >
                      <img
                        alt="flex-1 quotes icons"
                        className="flex-1 pl-10 pt-1 md:w-16 md:h-16 xl:w-20 xl:h-20 sm:pl-10 lg:w-20 lg:h-20 w-16 h-16 "
                        src="Quote.svg"
                      />
                      <img
                        alt="flex-1 quotes icons"
                        className="flex-1 pr-10 pt-1  md:w-16 md:h-16  lg:w-20 lg:h-20 xl:w-20 xl:h-20 w-16 h-16 sm:pr-10 "
                        src="Quote.svg"
                      />
                    </div>
                    <h1
                      className="pb-4 text-blue-500 sm:text-base md:text-base self-center sm:self-start xl:text-2xl lg:text-2xl   flex-1 items-start bold font-bold "
                    >
                      Person Name 4
                    </h1>
                    <p
                      className="flex-1 sm:left text-center sm:text-left leading-normal sm:leading-snug md:leading-snug lg:leading-loose xl:leading-loose lg:text-base xl:text-base md:text-sm sm:text-xs text-xs  w-4/5 xl:w-3/4	"
                    >
                      This website has helped me find my son, I will always be thankful for that.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        className="sc-bdnylx chowzb rec rec-arrow rec rec-arrow-right"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        tabIndex={0}
        type="button"
      >
        ‚ùØ
      </button>
    </div>
    <div
      className="sc-iCoHVE kRlYRh rec rec-pagination"
    >
      <button
        className="sc-gKAblj brcpxa rec rec-dot rec rec-dot_active"
        onClick={[Function]}
        tabIndex={-1}
        type="button"
      />
      <button
        className="sc-gKAblj ikyGAt rec rec-dot "
        onClick={[Function]}
        tabIndex={0}
        type="button"
      />
      <button
        className="sc-gKAblj ikyGAt rec rec-dot "
        onClick={[Function]}
        tabIndex={0}
        type="button"
      />
      <button
        className="sc-gKAblj ikyGAt rec rec-dot "
        onClick={[Function]}
        tabIndex={0}
        type="button"
      />
      <button
        className="sc-gKAblj ikyGAt rec rec-dot "
        onClick={[Function]}
        tabIndex={0}
        type="button"
      />
    </div>
  </div>
</div>
`;
